rudder-lang/libs/cfengine_core.rl

```diff
@@ -0,0 +1,345 @@
+# So we need to make sure our global context doesn't contain any variable having the same name + namespace
+# As of now the solution we found is to be proactive by explicitly preventing the user from adding any variable included in the list
+# This could also be useful in case a user wants to use any of these variables
+
+# Latest variable list update: CFEngine 3.13
+# Please keep track of the version when updating the list
+
+# `sys` global variables
+declare sys.arch
+declare sys.bindir
+declare sys.cdate
+declare sys.cf_promises
+declare sys.cf_version
+declare sys.cf_version_major
+declare sys.cf_version_minor
+declare sys.cf_version_patch
+declare sys.class
+declare sys.cpus
+declare sys.crontab
+declare sys.date
+declare sys.doc_root
+declare sys.domain
+declare sys.enterprise_version
+declare sys.expires
+declare sys.exports
+declare sys.failsafe_policy_path
+declare sys.flavor
+declare sys.flavour
+declare sys.fqhost
+declare sys.fstab
+declare sys.hardware_addresses
+declare sys.hardware_mac
+declare sys.host
+declare sys.inet
+declare sys.inet6
+declare sys.inputdir
+declare sys.interface
+declare sys.interfaces
+declare sys.interfaces_data
+declare sys.interface_flags
+declare sys.ip_addresses
+declare sys.ip2iface
+declare sys.ipv4
+declare sys.ipv4_1
+declare sys.ipv4_2
+declare sys.ipv4_3
+declare sys.key_digest
+declare sys.last_policy_update
+declare sys.libdir
+declare sys.local_libdir
+declare sys.logdir
+declare sys.license_owner
+declare sys.licenses_granted
+declare sys.long_arch
+declare sys.maildir
+declare sys.masterdir
+declare sys.nova_version
+declare sys.os
+declare sys.os_release
+declare sys.ostype
+declare sys.piddir
+declare sys.policy_entry_basename
+declare sys.policy_hub
+declare sys.policy_entry_dirname
+declare sys.policy_entry_filename
+declare sys.policy_hub
+declare sys.policy_hub_port
+declare sys.release
+declare sys.resolv
+declare sys.statedir
+declare sys.sysday
+declare sys.systime
+declare sys.update_policy_path
+declare sys.uptime
+declare sys.user_data
+declare sys.uqhost
+declare sys.version
+declare sys.windir
+declare sys.winprogdir
+declare sys.winprogdir86
+declare sys.winsysdir
+declare sys.workdir
+
+# `this` global variables
+declare this.bundle
+declare this.handle
+declare this.namespace
+declare this.promise_filename
+declare this.promise_dirname
+declare this.promise_linenumber
+declare this.promiser
+declare this.promiser_uid
+declare this.promiser_gid
+declare this.promiser_pid
+declare this.promiser_ppid
+declare this.service_policy
+declare this.this
+
+# `def` global variables
+declare def.jq
+
+# `edit` global variables
+declare edit.filename
+
+# `const` global variables
+declare const.dollar
+declare const.dirsep
+declare const.endl
+declare const.n
+declare const.r
+declare const.t
+
+# `connection` global variables
+declare connection.key
+declare connection.ip
+declare connection.hostname
+
+# `match` global variables
+declare match.0
```

Comment:
technically it can be any int value as it is used for regex captures

Reply:
Absolutely, I forgot about this one